### DISCUSSION TITLE
Suggested protocol for streaming a studio cam or DJ webcam into the station?

### DISCUSSION THREAD

**Discussion by Moonbase59** [savonet/liquidsoap#3179] [Q&A]
I think of letting a studio stream a studio cam into the station, or let a free DJ stream their webcam, to be included in the station’s video stream. This should be optional, i.e. only the video, the audio to be taken from the studio’s/DJ’s normal audio stream (typically generated by IDJC, SAM Broadcaster or mAirList).

**Which protocol would you suggest?** (Ideally based on actual experience.)

My plan is to re-use AzuraCast’s DJ authentication and set up an input harbor for this, with authentication. I’d use fallbacks to normally stream some video files available on the station VM, and whenever a studio/DJ connects, either let the stream from the input harbor take over, or have it display the studio cam as "picture-in-picture".

Personally, I feel using SRT would be the most professional way, but regarding bandwidth and ease-of-use, RTMP might be better (easy ffmpeg streaming from v4l2, OBS support).

It should work together with AzuraCast on cost-effective cloud VMs, so maybe on a 4 CPU, 4 GB VM, while the station would still be capable of running AzuraCast and producing an MP3, an AAC and a video stream via Icecast, maybe HLS or DASH later on. Req'd max resolution 720p25.

What do you think?

**Comment by Maddixe9** [comment]
exactly what i want to do... but clearly, (i have been reading everything i can find that u commented on!!) you ahead of me by a fair bit! hahah... but i would love to have a short convo with u about how u are getting along...basically, i am trying to set up my station/video stream and website on the cloud..i have only just started with ubuntu/docker/restreamer.. and i have been crashing my way thru..and i keep not getting all the way i want it... any chance u can provide me with some basic setup details..i.e one vm for each component/multiple vcn's? one massive tool of a server and docker containers? etc... but what i would love is... azuracast runnig the show..the ability for myself and my djs to remote in and stream video... a wordpress site to host the audio/video stream and be the front of the station..... thanks for all ya HINTS so far!! nice one!!
